costmap_2d/include/costmap_2d/layered_costmap.h

```
@@ -137,6 +137,12 @@ class LayeredCostmap
    * layers. */
   void setFootprint(const std::vector<geometry_msgs::Point>& footprint_spec);
 
+  /** @brief Utility method for enabling or disabling specific layers by name
+   * @param[in] layer_name The layer to enable or disable
+   * @param[in] enabled Whether the costmap layer should be enabled
+   *  */
+  void setLayerEnabled(const std::string &layer_name, const bool enabled);
```

In the context of recovery behaviours that are interacting with layers by name, it would be useful to see if a layer with a given name exists. If it doesn't then the recovery behaviour can fail fast and advance to the next one. Separate to seeing if one exists, getting the state of a named layer would be useful too. If a recovery behaviour is meant to disable a layer and do something but the layer is already disabled (perhaps the user was doing something?) We wouldn't want  reattempt something that has previously failed, also we don't want to set the state to something unexpected (enabled) in the cleanUp.